app/models/filter.rb

```diff
@@ -135,6 +141,21 @@ def expire_topbar_cache(sweeper)
     role.usergroups.each { |g| g.expire_topbar_cache(sweeper) }
   end
 
+  def disable_overriding!
+    self.override = false
+    self.save!
```

Comment:
Because AR saves has_many association immediately if the receiver is persisted, try `filter.organization_ids = []`, it immediately deletes all organization associations (watch SQL logger). This does not happen only if filter is new record, in such case it changes only in memory objects.